msal/tests/Test.MSAL.NET.Unit/AuthorityAliasesTests.cs

```diff
@@ -0,0 +1,202 @@
+                Url = string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/v2.0/.well-known/openid-configuration",
+                    TestConstants.ProdPrefNetworkEnv, TestConstants.Utid),
+                Method = HttpMethod.Get,
+                ResponseMessage = MockHelpers.CreateOpenIdConfigurationResponse(TestConstants.AuthorityUtidTenant)
+            });
+
+            // silent request targeting rt should find rt in cahce for authority with any environment alias
+            foreach (var envAlias in TestConstants.ProdEnvAliases)
+            {
+                HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler()
+                {
+                    Url = string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/oauth2/v2.0/token",
+                        TestConstants.ProdPrefNetworkEnv, TestConstants.Utid),
+                    Method = HttpMethod.Post,
+                    PostData = new Dictionary<string, string>()
+                    {
+                        {"grant_type", "refresh_token"}
+                    },
+                    // return not retriable status code
+                    ResponseMessage = MockHelpers.CreateInvalidGrantTokenResponseMessage()
+                });
+
+                try
+                {
+                    result = null;
+                    result = app.AcquireTokenSilentAsync(TestConstants.ScopeForAnotherResource,
+                        app.GetAccountsAsync().Result.First(),
+                        string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/", envAlias, TestConstants.Utid),
+                        false).Result;
+                }
+                catch (AggregateException ex)
+                {
+                    Assert.IsNotNull(ex.InnerException);
+                    Assert.IsTrue(ex.InnerException is MsalUiRequiredException);
+                }
+
+                Assert.IsNull(result);
+            }
+        }
+
+        private void ValidateCacheEntitiesEnvironment(TokenCache cache, string expectedEnvironment)
+        {
+            var accessTokens = cache.GetAllAccessTokensForClient(new RequestContext(new MsalLogger(Guid.NewGuid(), null)));
+            foreach (var at in accessTokens)
+            {
+                Assert.AreEqual(expectedEnvironment, at.Environment);
+            }
+
+            var refreshTokens = cache.GetAllRefreshTokensForClient(new RequestContext(new MsalLogger(Guid.NewGuid(), null)));
+            foreach (var rt in refreshTokens)
+            {
+                Assert.AreEqual(expectedEnvironment, rt.Environment);
+            }
+
+            var idTokens = cache.GetAllIdTokensForClient(new RequestContext(new MsalLogger(Guid.NewGuid(), null)));
+            foreach (var id in idTokens)
+            {
+                Assert.AreEqual(expectedEnvironment, id.Environment);
+            }
+
+            var accounts = cache.GetAllAccounts(new RequestContext(new MsalLogger(Guid.NewGuid(), null)));
```

Comment:
>new MsalLogger(Guid.NewGuid(), null) [](start = 67, length = 36)

can you use the same logger for all? What about the same request context?

Reply:
+1;

Reply:
On top of this, we should check that no errors are logged. This is quite important because a lot of the token cache logic does a try / catch / ignore exception and log, so it's the only way to tell if an exception was thrown...

Reply:
@SomskaPe ?

---
In reply to: [223489018](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1304#discussion_r223489018) [](ancestors = 223489018)

Reply:
You can setup a logger stub using NSubstitute and if anything is logged via Error or ErrorPii, fail the test. We do smth similar here [https://github.com:443/AzureAD/azure-activedirectory-library-for-dotnet.git/blob/dev/core/tests/Test.Microsoft.Identity.Core.Unit/CacheTests/CacheFallbackOperationsTests.cs#L188](https://github.com:443/AzureAD/azure-activedirectory-library-for-dotnet.git/blob/dev/core/tests/Test.Microsoft.Identity.Core.Unit/CacheTests/CacheFallbackOperationsTests.cs#L188)

---
In reply to: [224676750](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1304#discussion_r224676750) [](ancestors = 224676750,223489018)

Reply:
[https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/blob/dev/core/tests/Test.Microsoft.Identity.Core.Unit/CacheTests/CacheFallbackOperationsTests.cs#L188](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/blob/dev/core/tests/Test.Microsoft.Identity.Core.Unit/CacheTests/CacheFallbackOperationsTests.cs#L188)

---
In reply to: [224727703](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1304#discussion_r224727703) [](ancestors = 224727703,224676750,223489018)

Reply:
@  ?

---
In reply to: [224780601](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1304#discussion_r224780601) [](ancestors = 224780601,224727703,224676750,223489018)

Reply:
All your integration tests invoke logic from the CacheFallbackOperations class. All the methods in this class follow the pattern try { ... } catch all exception and log them. As such, all your integration tests should check that no errors or warnings are ever logged.

---
In reply to: [226488693](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1304#discussion_r226488693) [](ancestors = 226488693,224780601,224727703,224676750,223489018)